x-pack/plugins/index_management/server/index.ts

```diff
@@ -5,15 +5,16 @@
  * 2.0.
  */
 
-import { PluginInitializerContext } from 'src/core/server';
+import { PluginInitializerContext, PluginConfigDescriptor } from 'src/core/server';
 
 import { IndexMgmtServerPlugin } from './plugin';
 import { configSchema } from './config';
 
 export const plugin = (context: PluginInitializerContext) => new IndexMgmtServerPlugin(context);
 
-export const config = {
+export const config: PluginConfigDescriptor = {
   schema: configSchema,
+  deprecations: ({ deprecate }) => [deprecate('enabled', '8.0.0')],
```

Comment:
We'll need to add a `indexManagement.ui.enabled` settings so we can advise users to use that setting instead.